addons/project/models/project_task.py

```diff
@@ -1085,13 +1085,17 @@ def create(self, vals_list):
         # (portal) users that don't have write access can still create a task
         # in the project that will be checked using record rules
         new_context["default_create_in_project_id"] = default_project_id
+        is_portal = self.env.user._is_portal() and not self.env.su
+        if is_portal:
+            default_milestone_id = new_context.pop('default_milestone_id', False)
         if not self._has_field_access(self._fields['user_ids'], 'write'):
             # remove user_ids if we have no access to it
             new_context.pop('default_user_ids', False)
         self = self.with_context(new_context)
 
         self.browse().check_access('create')
         default_stage = dict()
+        child_ids_list = []
```

Comment:
We can manage adding the subtask in additional_vals. but, the test_create_task_in_project_sharing test case may fail. Please check if possible. otherwise, it's okay.